### ISSUE TITLE
Indentation of inheritance with parameters

### ISSUE THREAD

**Issue by mklnwt** [pinterest/ktlint#2661]
Indentation for inheritance implementation does not look correct.

For example
```Kotlin
class Bar(
    baz: String,
) : Foo(
        baz = baz,
    ) {
    fun qux() { /* ... */ }
}
```

## Expected Behavior
```Kotlin
@Test
fun `Given a class with inheritance`() {
    val code =
        """
        class Bar(
            baz: String,
        ) : Foo(
            baz = baz,
        )
        """.trimIndent()
    indentationRuleAssertThat(code).hasNoLintViolations()
}
```

## Current Behavior
```Kotlin
@Test
fun `Given a class with inheritance`() {
    val code =
        """
        class Bar(
            baz: String,
        ) : Foo(
                baz = baz,
            )
        """.trimIndent()
    indentationRuleAssertThat(code).hasNoLintViolations()
}
```

## Additional information
* Current version of ktlint: 1.2.1

**Comment by paul-dingemans** [comment]
Code style `ktlint_official` takes a different perspective than you might be used. From a consistency viewpoint the new format allows better formatting in case the class extends a super class having arguments as well as some additional interfaces. Code below is consistently formatted:
```
class Foo(
    foo1: Foo1,
    foo2: Foo2,
) : Foobar1(
        "foobar1",
        "foobar2",
    ),
    Foobar2,
    Foobar3 {
    fun foo() = "foo"
}
```
The super class `Foobar1` and interfaces `Foobar2` and `Foobar3` are all aligned without the closing parenthesis and comma of `Foobar1` disturbing the formatting.

With the other code styles, the old way of formatting is still available:
```
class Foo(
    foo1: Foo1,
    foo2: Foo2
) : Foobar1(
    "foobar1",
    "foobar2"
),
    FooBar2,
    FooBar3 {
    fun foo() = "foo"
}
```

Code style `ktlint_official` will always choose the most consistent formatting possible. If might take some time to get used to.

**Comment by mklnwt** [comment]
Considering that function calls look like this:
```Kotlin
foos.map {
    it + 1
}.bar()
    .baz()
```
or
``` Kotlin
foo(
    bar = "bar",
).bar()
    .baz()
```
I would say that the inheritance example is not fully consistent.

Otherwise is it possible to use `ktlint_official`, but switch this specific rule part to the old style?

**Comment by paul-dingemans** [comment]
I think we have had discussions like this before. You have to disclose your `.editorconfig` settings. Above is most likely the result of disabling certains rules or use non-default settings. 

With `.editorconfig` below, the code above is not accepted, and it is formatted consistently.
```
root = true

[*.{kt,kts}]
ktlint_code_style = ktlint_official
ktlint_standard = enabled
ktlint_experimental = enabled
```

> Otherwise is it possible to use ktlint_official, but switch this specific rule part to the old style?

No that is not supported, and will [not be supported](https://pinterest.github.io/ktlint/latest/faq/#can-a-new-toggle-be-added-to-optionally-enabledisable-format-code-in-a-particular-way).

**Comment by mklnwt** [comment]
@paul-dingemans I have mostly worked with the unit test provided above.

I experimented a bit locally and it seems that a missing experimental rule is the reason the join to the previous line is not enforced.

I still believe that the indentation rule by itself (deactivated `Expected single space before the first super type (standard:class-signature)`) should not produce the example above.

**Comment by paul-dingemans** [comment]
> Sidenote and personal opinion:
> Breaking after the assignment AND at the first method call seems very inelegant and wasting a lot of space (both vertically and horizontally).

With short identifiers it seems indeed a waste of space. In real life, identifiers can also become (very) long. 

```kotlin
val foos =
    foosfoosfoosfoosfoosfoosfoosfoosfoosfoosfoosfoos
        .foofoofoo()
        .bar()
        .baz()
        .qux()
```
which makes `foofoofoo` harder to discover
```kotlin
val foos = foosfoosfoosfoosfoosfoosfoosfoosfoosfoosfoosfoos.foofoofoo()
    .bar()
    .baz()
    .qux()
```

As with a lot of decisions regarding formatting it is subjective what reads better. I always try to look at extreme examples when to decide what format to use.
